src/pkg/podman/podman.go

```diff
@@ -226,18 +237,80 @@ func IsToolboxImage(image string) (bool, error) {
 	return true, nil
 }
 
-// Pull pulls an image
-func Pull(imageName string) error {
+func Login(registry string, extraArgs ...string) error {
+	var stderr bytes.Buffer
+
 	logLevelString := LogLevel.String()
-	args := []string{"--log-level", logLevelString, "pull", imageName}
+	args := []string{"--log-level", logLevelString, "login", registry}
+	args = append(args, extraArgs...)
 
-	if err := shell.Run("podman", nil, nil, nil, args...); err != nil {
+	if err := shell.Run("podman", os.Stdin, os.Stdout, &stderr, args...); err != nil {
+		if stderr.Len() == 0 {
+			return fmt.Errorf("failed to login to %s: %w", registry, err)
+		}
+
+		err := parseErrorMsg(&stderr).(*internalError)
 		return err
 	}
 
 	return nil
 }
 
+// Pull pulls an image. Wraps around command 'podman pull'.
+//
+// The image pull can fail for many reasons. The recognized reasons are:
+//   - manifest unknown (ErrManifestUnknown)
+//   - name unknown (ErrNameUnknown)
+//   - unauthorized (ErrUnauthorized)
+//
+// To learn more about OCI API error codes, see: https://github.com/opencontainers/distribution-spec/blob/main/spec.md
```

Comment:
I am a bit confused about what I can learn from this URL. :)

Is it about the section titled *Error Codes*? It talks about the `code` field which must contain only uppercase characters and underscores, but that's not what we are looking for in the code below. The error codes might be defined by OCI, but that doesn't help us much since we are just scraping `stderr` for known sub-strings.

Reply:
You're right. I removed the link.